core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -483,9 +485,20 @@ object Parser extends ParserInstances {
           }
           .toList
 
-      if (rangeMerge.isEmpty) errors.distinct.sorted
+      // we don't need Fails that point to duplicate offsets
+      val nonFailOffsets: Set[Int] =
+        el.iterator.filterNot(_.isInstanceOf[Fail]).map(_.offset).toSet
+
+      val errors1 = NonEmptyList.fromListUnsafe(
```

Comment:
note, this still has to have at least one item, if there were only one item, there can't be a duplicate, so nothing is filtered out.